Expense/expenseRouter.js

```diff
@@ -0,0 +1,32 @@
+const express = require("express");
+const Budget = require('../Budget/Budget.js');
+const Category = require('../Category/Category.js');
+const Expense = require('./Expense.js');
+const router = express.Router();
+const mongoose = require("mongoose");
+
+router.route('/').post(async (req, res) => {
```

Comment:
We're missing a key element here for the post function, saving refs to the Expense instance.

Here's an example of saving a Category ref to an Expense:

```
router.route('/').post(async (req, res) => {
    const { amount, description, category } = req.body
    try {
        const response = await Category.find({ title: category }).select('_id')

        const expense = new Expense({
            amount,
            category: response[0]._id,
            description
        })
        expense.save((err, post) => {
            if (err) return next(err)
            res.status(201).json(expense)
        })
    } catch (error){
        res.status(500).json({error: error});
    }
})
```

You could do the same with Budget. Just make sure you adjust your `req.body` accordingly:
```
{
    "amount": 210,
    "description": "description",
    "category": "Food/Dining"
}
```

And that _that_ category actually exists in your DB.